src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -322,6 +323,61 @@ export function VislibVisualizationsPieChartProvider(Private) {
       }
     }
 
+    readableNumber(number) {
+      let tier = Math.log10(Math.abs(number)) / 3 | 0;
+      if(tier === 0) return number;
+      if(tier > 4) tier = 4;
+      const postfix = this.ABBREVIATIONS[tier];
+      const scale = Math.pow(10, tier * 3);
+      const scaled = number / scale;
+      let formatted = String(scaled.toFixed(1));
+      if (/\.0$/.test(formatted)) {
+        formatted = formatted.substr(0, formatted.length - 2);
+      }
+      return formatted + postfix;
+    }
+
+    addDonutDescriptor(svg, width, height) {
+      const self = this;
+      const metricLabel = self.events.handler.data.data
+        .raw.columns.find(function (element) {
+          return element.categoryName === 'metric';
+        }).label;
+      const descriptor = svg.append('text')
+        .style('text-anchor', 'middle')
+        .attr('font-size', function (chartData) {
+          const dividend = Math.min(width, height);
+          let length = 0;
+          if(chartData.raw) {
+            length = chartData.raw.columns.length;
+          }
+          const divisor = Math.max(length, 4);
```

Comment:
Could you please explain (and ideally add a comment) what this part does exactly? Why are we using `4` here as a minimum?

Reply:
This code determines the size of font for descriptor.  As we add sub-buckets, it increases divisor to accommodate more pie layers. 4 is the minimum number derived by trial and error, the divisor should be set to. Would you have any suggestion for a better logic to determine font size (It has to be dynamic, based on size of the panel and number of sub-buckets)?